src/components/Settings/Input/index.tsx

```diff
@@ -0,0 +1,46 @@
+import styled from 'styled-components/macro'
+
+import Row from '../../Row'
+
+export const Input = styled.input`
+  width: 100%;
+  display: flex;
+  flex: 1;
+  font-size: 16px;
+  border: 0;
+  outline: none;
+  background: transparent;
+  text-align: right;
+  &::-webkit-outer-spin-button,
+  &::-webkit-inner-spin-button {
+    -webkit-appearance: none;
+  }
+  ::placeholder {
+    color: ${({ theme }) => theme.textTertiary};
+  }
+`
+
+export const InputContainer = styled(Row)<{ error?: boolean }>`
+  padding: 8px 16px;
+  border-radius: 12px;
+  width: auto;
+  flex: 1;
+  input {
+    color: ${({ theme, error }) => (error ? theme.accentFailure : theme.textPrimary)};
+  }
+  border: 1px solid ${({ theme, error }) => (error ? theme.accentFailure : theme.deprecated_bg3)};
+  ${({ theme, error }) =>
+    error
+      ? `
+        border: 1px solid ${theme.accentFailure};
+        :focus-within {
+          border-color: ${theme.accentFailureSoft};
+        }
+      `
+      : `
+        border: 1px solid ${theme.backgroundOutline};
+        :focus-within {
+          border-color: ${theme.accentActiveSoft};
+        }
+      `}
```

Comment:
These should use the `css` macro

Reply:
Why would you use a macro in this case? We don't wrap it with `css` when we do something like this:
```
${({ theme, error }) => border: 1px solid ${theme.accentFailure}
```
so I am not sure what is different about two properties. I personally use `css` when I want to reuse some styling across components, but for inline definitions like this - I don't.

Reply:
https://styled-components.com/docs/api#css